app/src/main/java/woowacourse/shopping/domain/usecase/AddToCartUseCase.kt

```
@@ -0,0 +1,30 @@
+
+            if (item != null) {
+                val updatedResult =
+                    cartRepository.changeCartItem(productId, item.increase(quantity).quantity)
+                when (updatedResult) {
+                    is UpdateItemResult.Success -> AddItemResult.Incremented(updatedResult.cart)
+                    is UpdateItemResult.Error -> AddItemResult.Error(updatedResult.message)
+                }
+            } else {
+                cartRepository.addItem(productId, quantity)
+            }
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)